### ISSUE TITLE
[Feature Request] Upload/choose vocabulary sets

### ISSUE THREAD

**Issue by Sinbu** [baileysnyder/japanese-conjugation#10]
I don’t know common formats, but something to take from like anki or other places and be able to select what you want to train. Will likely look at this myself in a bit and propose a solution (writing a converter function for common types / creating config to select which sets to drill).

**Comment by baileysnyder** [comment]
Interesting idea, and ty for the compliment :] To avoid working against the conjugation function already in place, it'd probably be easiest to make new functions that take in some word data and convert them into objects following the format in worddata.js. You'd still need to be careful of sizing issues where things might look weird with words or English definitions that are too long, but that shouldn't be a big deal.

Were you thinking along the lines of people adding individual words (then maybe pulling the extra data from somewhere), uploading lists of words in a particular format, or just giving them a dropdown of a wider variety of word lists to choose from?

My biggest concern is adding complexity that doesn't help the average user. While I could see some use in learning conjugations for a particular word causing you trouble, I think the main premise of the app is that you can learn general conjugation rules and the specific words don't matter as much. I think one thing this app has over other similar apps is its simplicity, so I wouldn't want to add anything that a user needs to decipher in the options if it's not going to help them.

**Comment by Sinbu** [comment]
Absolutely agree that the scope would be simply to help people add words into worddata.js, then having a configuration. The first version could simply be “upload your own list of words as an override for this session” and then build from there if there is more useful stuff.

I agree around unnecessary complexity. The main reason for me is that I have trouble learning vocabulary unless I actually use it and conjugate it properly, so if I could upload 15 words I’m drilling daily into this, then just power through, it would help me immensely. I think it would help others too. If not, I can simply just download a local copy and modify the worddata.js file :).

Tl;dr — MVP version that would allow people to overwrite worddata.js with a list of their own words (temporarily).

**Comment by gialmisi** [comment]
I was thinking of maybe fetching verbs from the WaniKani API, so that people could practice conjugating verbs they have learned on the WK. Should doable. Granted I have not looked much into the code yet, I just came across this repo. Maybe I will give it a shot.
